netmiko/cisco/cisco_vxr_ssh.py

```diff
@@ -0,0 +1,309 @@
+        start_time = time.time()
+        current_time = time.time()
+
+        # Keep reading data until search_pattern is found or session is alive or max_read_timeout is reached
+        while current_time - start_time < self.max_read_timeout and self.is_alive():
+            new_data = self.read_channel()
+            if new_data:
+                if self.ansi_escape_codes:
+                    new_data = self.strip_ansi_escape_codes(new_data)
+                # Replace Null Characters sent while checking session (alive) status
+                new_data = new_data.replace("^@", "")
+
+                output += new_data
+                try:
+                    lines = output.split(self.RETURN)
+                    first_line = lines[0]
+                    # First line is the echo line containing the command. In certain situations
+                    # it gets repainted and needs filtered
+                    if BACKSPACE_CHAR in first_line:
+                        pattern = search_pattern + r'.*$'
+                        first_line = re.sub(pattern, repl='', string=first_line)
+                        lines[0] = first_line
+                        output = self.RETURN.join(lines)
+                except IndexError:
+                    pass
+                if re.search(search_pattern, output):
+                    break
+
+                if re.search(config_large_msg, output):
+                    output = self.send_command(command_string=self.RETURN,
+                                               auto_find_prompt=False, strip_prompt=False, strip_command=False, )
+                    output += self.read_channel()
+                    if re.search(search_pattern, output):
+                        break
+            else:
+                time.sleep(loop_delay)
+            log.info("Pattern not found. Time waited: {}".format(current_time - start_time))
+            current_time = time.time()
+
+        else:  # nobreak
+            if not self.is_alive():
+                raise IOError(
+                    "Session went down while checking for prompt after sending command.\nSearch pattern: {}".format(search_pattern))
+            else:
+                if expect_string is None:
+                    raise IOError("Prompt not found after sending command and waiting for {} seconds.\nExpected Prompt: {}.\nOutput: {}".format(
+                        self.max_read_timeout, search_pattern, output))
+                else:
+                    raise IOError("Search Pattern not found after sending command and waiting for {} seconds.\nExpected Prompt: {}\nOutput: {}".format(
+                        self.max_read_timeout, search_pattern, output))
+        if current_time - start_time >= 10:
+            log.info("Command took {} seconds".format(current_time - start_time))
+        output = self._sanitize_output(output, strip_command=strip_command,
+                                       command_string=command_string, strip_prompt=strip_prompt)
```

Comment:
This can be moved inside the `while`'s `else` condition and `current_time` if replaced by`time.time()` would lead to insignificant time difference between actual loop's end time and this time (assuming the original intent to maintain `current_time` variable)

Reply:
same as earlier. current_time variable is used to avoid calculating time.time() again and again because its used in logs

Reply:
for santize_output: its only needed once we have got the full output, hence moving it while's else condition wont work